### ISSUE TITLE
ASFs can use plasma rifles for strafing runs

### ISSUE THREAD

**Issue by EnbyMyu** [MegaMek/megamek#5232]
### Environment
MekHQ/MegaMek 0.49.18
Java Vendor: Eclipse Adoptium
Java Version: 21
Win11 23H2

### Description
Aerospace Fighters with plasma rifles can use them for strafing runs. Presumably since they are classified as 'energy' weapons. But being ammo-dependent I was told to open a bug report.

To reproduce, create an ASF with plasma rifles and perform a strafing run as normal.

![image](https://github.com/MegaMek/megamek/assets/159649649/39b980f1-c7d4-47c1-be17-40eb00fbb0c8)

### Files
[plasmastrafe.sav.gz](https://github.com/MegaMek/megamek/files/14532174/plasmastrafe.sav.gz)

Team name is "The Wrecking Belles"

**Comment by Sleet01** [comment]
There's probably also an AS conversion issue for these; if I don't pick this up, whoever does will want to make sure that Plasma Rifles properly prevent the ENE keyword for AS conversion.

**Comment by RaozSpaz** [comment]
Wouldnt they still get the ENE tag as the Plasma Rifle ammo is inert.

**Comment by Sleet01** [comment]
> Wouldnt they still get the ENE tag as the Plasma Rifle ammo is inert.

Negatory:
```
Energy (ENE)

A unit with this ability does not rely on ammunition-fed 
weaponry or fuels, ...

Conversion: This special is assigned to any non-infantry 
unit type that carries no explosive components. Explosive 
components that disqualify units from receiving this ability 
include all weapons that use ammunition (including
... 
plasma weapons....```

**Comment by RaozSpaz** [comment]
Its weird that plasma weapons would be included under a list of explosive components. I thought plasma ammo didn't explode. Nor the weapon. That might be a rules question.

**Comment by Sleet01** [comment]
> Its weird that plasma weapons would be included under a list of explosive components. I thought plasma ammo didn't explode. Nor the weapon. That might be a rules question.
> 
> [Apologies for getting off topic]

Yeah, it was a surprise to me initially, too.  I actually tracked one of the devs down on another server and got it confirmed, while looking at some other AS conversion code.
